websubhub-ballerina/build.gradle

```diff
@@ -127,19 +151,52 @@ task revertTomlFile {
     }
 }
 
+task ballerinaTest {
+    dependsOn(updateTomlFile)
+    dependsOn(initializeVariables)
+    finalizedBy(revertTomlFile)
+
+    doLast {
+        exec {
+            workingDir project.projectDir
+            environment "JAVA_OPTS", "-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true"
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat test --code-coverage --includes=* ${groupParams} ${disableGroups} ${debugParams} && exit %%ERRORLEVEL%%"
+            } else {
+                commandLine 'sh', '-c', "$balJavaDebugParam $distributionBinPath/bal test --code-coverage --includes=* ${groupParams} ${disableGroups} ${debugParams}"
+            }
+        }
+    }
+}
+
+test {
+    dependsOn(ballerinaTest)
+}
+
 task ballerinaBuild {
     inputs.dir file(project.projectDir)
+    dependsOn(initializeVariables)
     finalizedBy(revertTomlFile)
```

Comment:
It's better if we put the `dependsOn(updateTomlFile)` here since we have the `finalizedBy(revertTomlFile)` here. It will increase the readability.

Alternatively, we can move this to the line no 284. But it's better if we keep a single convention for this. (In the `ballerinaTest` task and in the `ballerinaBuild` task)